### PR TITLE
EnginkoFix

### PR DESCRIPTION
**Pull request by J2thatsme**
Sensor now sends all samples and emits on defined timestamps

### CODE REVIEW

File: types/enginko/MCF-LW12CO2E/uplink.js [hunk 1]
```diff
@@ -783,16 +825,13 @@ function parseReportData(payload) {
...
-
-function isNumeric(str) {
-  return /^\-?[0-9]+(e[0-9]+)?(\.[0-9]+)?$/.test(str);
+  return content;
 }
 
 function consume(event) {
   const payload = event.data.payloadHex;
   const sample = {};
+  const sample2 = {};
```
Comment (mariusgiger) [inline types/enginko/MCF-LW12CO2E/uplink.js:834]: why do you need to resulting samples?

Comment (J2thatsme) [reply types/enginko/MCF-LW12CO2E/uplink.js:834]: The sensor can send two samples with the same structure at the same time. The only difference being that they're 10 minutes apart. Thats why i need two.

Comment (mariusgiger) [reply types/enginko/MCF-LW12CO2E/uplink.js:834]: please try to make this more generic, something along the lines of 
```
emit("sample", parseSample(event.data.payloadHex[0:50]));
emit("sample", parseSample(event.data.payloadHex[50:79]));
```

File: types/enginko/MCF-LW12CO2E/uplink.js [hunk 2]
```diff
@@ -838,12 +876,35 @@ function consume(event) {
     if (isNumeric(value)) {
       value = Number(value);
     }
-    sample[variable] = value;
+
+    // If theres a second date, it implicates that a second sample will get emited
+    if (
```
Comment (mariusgiger) [inline types/enginko/MCF-LW12CO2E/uplink.js:881]: the logic still seems a bit confusing to me - I would already decide when to create new samples in the for loop looping over content. basically each time you see a date, you will need a new sample

Comment (J2thatsme) [reply types/enginko/MCF-LW12CO2E/uplink.js]: Not all samples have a date so i need to check for that too ^^' . Hope i coded it clearer now

Comment (mariusgiger) [reply types/enginko/MCF-LW12CO2E/uplink.js]: what is the format of the message if the second sample does not have a date? how can this happen..? ^^

**Review by mariusgiger — APPROVED** [review]

**Pull request merged**